datanode/entities/pagination.go

```diff
@@ -111,7 +111,7 @@ func NewCursorPagination(first *int32, after *string, last *int32, before *strin
 }
 
 func CursorPaginationFromProto(cp *v2.Pagination) (CursorPagination, error) {
-	if cp == nil || (cp != nil && cp.First == nil && cp.Last == nil) {
+	if cp == nil || (cp != nil && cp.First == nil && cp.Last == nil && cp.After == nil && cp.Before == nil) {
```

Comment:
Can we please have a unittest for this? Because this line of code is now being changed for the third time now, it would be good to make sure its right this time:
https://github.com/vegaprotocol/vega/commit/eb53bc486b224767fac66eac19dd5e3121b69649
https://github.com/vegaprotocol/vega/commit/0d30f246f470cdd4accf6fff5163ce7d20f70951